### ISSUE TITLE
[Update Request]: KDiff3 still at version 1.9.6

### ISSUE THREAD

**Issue by cheesecracker** [microsoft/winget-pkgs#107993]
### What type of update are you requesting?

A new version of an existing package

### Current Package Identifier

KDE.KDiff3

### Package Version

1.10.4

### Please describe the changes you would like to see

STEPS TO REPRODUCE
1. Run "winget install -e --id KDE.KDiff3" in PowerShell

OBSERVED RESULT
Version 1.9.6 is installed

EXPECTED RESULT
Version 1.10.4 is installed

ADDITIONAL INFORMATION
Question:
  Is KDiff3 inside *winget store* still maintained?
  Do we (layman users) need to file such request for every new version of this application?

If it's not maintained, may I suggest you to *delete* everything about KDiff3 in winget store.  In this way, people would be forced to download the latest version from your web site.  This could eliminate a lot of potential problems.

However, if it's still maintained, please fix the issue.  Thanks in advance.

**Comment by Trenly** [comment]
This package might not be able to be updated because the installer seems to always throw this prompt -
![image](https://github.com/microsoft/winget-pkgs/assets/12611259/2bb5cb04-6a2c-440e-807d-00bb426242fe)

**Comment by cheesecracker** [comment]
I'm a simple **end-user**.  I know nothing about the making of package.

FYI, a few hours ago, I posted a bug report to KDE about this issue:
https://bugs.kde.org/show_bug.cgi?id=470221

That bug was immediately closed by them, saying that they "don't have any part in the maintenance or access to the store".

Back to your comment about the "throwing the prompt", in my perspective as an end-user, I would say:
Could the "mechanism" be programmed to detect pre-1.8 versions?  If pre-1.8 version is installed, is it possible to just display a message, through PowerShell of course, to tell the user that KDiff3 has to be uninstalled (manually) first?

If **nobody** is going to take care / maintain this package, please delete everything in the package.

**Comment by Trenly** [comment]
> That bug was immediately closed by them, saying that they "don't have any part in the maintenance or access to the store".

Correct. The winget source is maintained by the community. While publishers can submit their own updates, most packages are submitted by the community

> Back to your comment about the "throwing the prompt", in my perspective as an end-user, I would say:
> Could the "mechanism" be programmed to detect pre-1.8 versions? If pre-1.8 version is installed, is it possible to just display a message, through PowerShell of course, to tell the user that KDiff3 has to be uninstalled (manually) first?

Not exactly. The prompt is thrown even on a fresh install of windows. Since the prompt is part of the KDE installer, that is most likey either a bug or an intentional design choice by KDE. It is a requirement that all packages in the winget source must be able to be installed without user interaction, which makes this a blocking issue unless 1) there is a way to suppress the message box, 2) KDE removes or fixes the message box, or 3) winget implements filtering by installation mode.

As to your idea around winget detecting pre-1.8 versions, there is a feature request open for that type of behavior (see https://github.com/microsoft/winget-cli/issues/3007). However, since the prompt is shown even when KDiff3 has never been installed, there would be no difference even if we could detect it.

> If nobody is going to take care / maintain this package, please delete everything in the package.

Some packages are quite popular and are maintained by users regularly. Others are not as commonly used and a request like this is needed to inform a contributor that a newer version is available. Thats one of the downsides of being a community-driven repository. However, thats not to say "nobody" is maintaining it; rather - we are trying to maintain it but the popup presents a blocking issue.



-----

> I'm a simple end-user. I know nothing about the making of package.

The best way I can explain it is that there are package publishers (Like KDE or Google or Zoom) and there are winget contributors, who are just end users like you or me.  

A publisher actually creates the software. They create the installer, the uninstaller, the actual application that runs. Anything that has to do with the behavior of the application is the responsibility of the publisher.

A winget contributor submits a data file called a manifest to this repository, the winget source. The data file only tells winget how to download the installer the publisher created, if any special arguments are needed for an unattended install, and some information needed to detect the software is installed. 

All winget does is download the installer from the publishers website and run it. For an uninstall, it just calls the program's default uninstaller.

**Comment by cheesecracker** [comment]
@Trenly
Thanks for your explanations on making the package.  Really very very appreciate it.  But I'm afraid you have wasted your time on me.  You see, I have my job and family.  I don't quite have spare time.  Writing bug reports like this is already a large effort from me.  The information you gave me, I'm not going to use it anywhere, unfortunately.  I'm not going to make manifest file, sure of that.

I see that the setup-making application for KDiff3 is NSIS.  I think maybe you could look into the NSIS files to understand how to get around it.

For the time being, I'm going to forget about winget, download the setup and just install it in the old way.

Good luck to you, guys. 👍

**Comment by stephengillie** [comment]
> I'm a simple **end-user**. I know nothing about the making of package.
> 
> FYI, a few hours ago, I posted a bug report to KDE about this issue: https://bugs.kde.org/show_bug.cgi?id=470221
> 
> That bug was immediately closed by them, saying that they "don't have any part in the maintenance or access to the store".

This is unfortunately a very common way for open-source projects to handle bug reports - closing with extreme prejudice. In my personal account and on my own time, this kind of passive aggression is a major reason why I haven't contributed to most open-source projects, despite even some YouTube videos begging viewers to do so. The requirement that a person spend months or years learning both the repo history and the behaviors of the major contributors is too high of a barrier to entry for the majority of users to surmount, just to make one suggestion to one codebase. One of my goals in this role is to at least provide closure and background info, especially to newer users, and help to offset the trend for excessively succinct closures.

**Comment by reporter123** [comment]
The user interaction has been amend to honor the silent switch. That was an oversight on my part.  Unfortunately I rely craft for the windows build and thus part of the install script is beyond my control. This prevents me from detecting if the older version was present as craft generate script currently runs the uninstaller for that version before I have a chance to do anything. 

The patched install should be available under  kdiff3-1.10.4-slientfixed-windows-x86_64.exe in the next 24 to 48 hrs. 1.10.5 will include the change.

**Comment by cheesecracker** [comment]
Original reporter here:

I have just done a "winget install -e --id KDE.KDiff3" in PowerShell (in a clean Windows 10).

Result: version 1.12.2 is installed.

I think you could close this issue.

**Comment by ItzLevvie** [comment]
Close with reason: I am not able to reproduce this issue on the latest version of Windows 11 and WinGet;
